packages/serverless-components/nextjs-component/src/component.ts

```diff
@@ -446,8 +451,14 @@ class NextjsComponent extends Component {
       return inputValue[lambdaType] || defaultValue;
     };
 
+    const queue = await sqs({
```

Comment:
BTW, I guess ISR is a build time feature, so would you be able to detect if users are using it and only create the SQS queue and regeneration Lambda if so?

Reply:
Technically there's no way to know if a dynamic SSG page with fallback true/blocking will use ISR if it does not prerender any static paths that do.

But in practice it should be good enough to only enable regeneration when there are some revalidation times in the prerender manifest. It would be easy to work around.

Reply:
I've added a check to the serverless components and the cdk construct for the presence of ISR pages, and create a queue based on that. I agree, I think it'd be easy enough to work around in case you do need the queue.